### DISCUSSION TITLE
What causes the node state to become disabled?

### DISCUSSION THREAD

**Discussion by rprj** [openthread/openthread#7934] [Q&A] [answered]
Hello,

I have some full-thread devices that have been running for days, but sometimes their node state becomes **disabled**. This happens to some intermittently, but for some devices, they never recover back to either child, router, or leader state. What could be some of the causes for this?

**Comment by jwhui** [comment, marked as answer]
By "disabled", I assume you are referring to `OT_DEVICE_ROLE_DISABLED`.

OpenThread does not enter `OT_DEVICE_ROLE_DISABLED` on its own, except on initialization. As a result, either some application code is calling an API to disable OpenThread or the device is unexpectedly resetting.

Increasing the log level and capturing the logs can help determine the cause.
